### PR TITLE
Publications

### PR DESCRIPTION
**Pull request by DarekMazur**

### CODE REVIEW

File: src/lib/helpers/validateISSN.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+export const validateISSN = (issn: string) => {
+  const issnPattern = /^[0-9]{4}-?[0-9]{3}[0-9X]$/;
+  if (!issnPattern.test(issn.trim())) {
+    return false;
+  }
+
+  const cleanISSN = issn.trim().toUpperCase();
+
+  const digitsOnly = cleanISSN.replace('-', '');
```
Comment (DarekMazur) [inline src/lib/helpers/validateISSN.ts]: Replacer removes **only first** dash

File: src/lib/helpers/__test__/validateISSN.test.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const validISSN = ['0317-8471', '03178471', '2434-561X'];
+
+const invalidISSN = [
+  '2434-561x', // incorrect lowercase X
+  '123-4567', // invalid - too short - format
+  'ABCD-EFGH', // invalid characters
+  '0317-8472', // invalid checksum
+  12345678, // param is not string
+];
+
```
Comment (DarekMazur) [inline src/lib/helpers/__test__/validateISSN.test.ts:12]: Missing edge cases, ie only zeros ('0000-0000'), only spaces ( '       ') or empty string ('')

File: src/mocks/browser.ts [hunk 3]
```diff
@@ -21,6 +22,41 @@ const generateAlertBlock = (type: string): string => {
...
+  return digits;
+};
+
+const generateISSN = () => {
+  const baseDigits = generateBaseDigits();
+  const checkDigit = calculateCheckDigit(baseDigits);
+  const formattedBase = baseDigits.join('').replace(/^(\d{4})(\d{3})$/, '$1-$2');
+  return `${formattedBase}${checkDigit}`;
+};
+
```
Comment (DarekMazur) [inline src/mocks/browser.ts:59]: Concern to create helpers

File: src/mocks/db.ts [hunk 4]
```diff
@@ -75,4 +75,14 @@ export const db = factory({
...
+  publication: {
+    id: primaryKey(faker.string.uuid),
+    title: () => faker.lorem.words({ min: 1, max: 4 }),
+    author: () => faker.person.fullName(),
+    publisher: oneOf('publisher'),
+    year: () => faker.date.past().getFullYear().toString(),
+    isbn: nullable(() => faker.number.int().toString()),
+    issn: nullable(() => faker.number.int().toString()),
+    description: () => faker.lorem.paragraph(),
+  },
```
Comment (DarekMazur) [inline src/mocks/db.ts:87]: Missing publication type or interface

File: src/mocks/browser.ts (hunk 3 above)
Comment (DarekMazur) [reply src/mocks/browser.ts:59]: For now, these methods are only used to mock data
